src/helpers/calculations.js

```diff
@@ -0,0 +1,67 @@
+import { formatDuration, add, isAfter } from 'date-fns';
+
+export function getLifeCount(date, maxAge = 90) {
+  if (!(date instanceof Date) || isNaN(date)) {
+    throw new TypeError(
+      'You should specify correct date object in first parameter'
+    );
+  } else if (
+    isAfter(date, new Date()) ||
+    isAfter(add(new Date(), { years: -maxAge }), date)
+  ) {
+    throw new Error('Seems like you have a time machine');
+  }
+
+  const ageMiliseconds = Date.now() - date.getTime();
+  const lifeMiliseconds = maxAge * 365.25 * 1000 * 3600 * 24 + 1000 * 3600 * 24;
+
+  const ageValues = parseTime(ageMiliseconds);
```

Comment:
I think all or most of calculations could be replaced with date-fns specific operators.
here is a case for today, I'm trying my b-date
```javascript
test("life count returns correct object for custom date", () => {
  expect(getLifeCount(new Date("1985-07-19"), 90)).toMatchObject({
    years: [34, 90],
    months: [419, 1080],
    weeks: [1822, 4696],
    days: [12754, 32873],
    percent: (12754 / 32873) * 100,
    remain: "55 years, 4 weeks",
    lived: "34 years, 11 months, 1 day"
  });
});
```
And it's passing(I just copied what function returned me), but values are not correct.
I haven't lived yet 34 years, 11 months, 1 day. this will be tomorrow :)
So it's a rounding issue I guess.
Here is a proposal for a solution (just proposal, you can tune your own, and I'm not sure that date-fns will provide you this)
1. after validity checks create a `duration`
2. extract from it required numbers years, months, weeks and days
3. convert maxAge (90) to the values months, weeks and days (no lib required)
4. compose data for return (remain with formatDuration, lived - from another duration from b-daty to today)

Reply:
Service that you've shown me (https://www.timeanddate.com/date/duration.html) has a checkbox to include one day
![image](https://user-images.githubusercontent.com/37845039/85123631-43de3880-b231-11ea-95aa-9bfcb087c48c.png)
So I think we can just subtract one day from the date

Reply:
Because our service includes "today" date in calculations

Reply:
Okay )